packages/x-charts/src/BarChart/BarElement.tsx

```diff
@@ -55,7 +55,8 @@ export const BarElementPath = styled(animated.rect, {
   overridesResolver: (_, styles) => styles.root,
 })<{ ownerState: BarElementOwnerState }>(({ ownerState }) => ({
   stroke: 'none',
-  shapeRendering: 'crispEdges',
+  // this makes curves look bad
+  // shapeRendering: 'crispEdges',
```

Comment:
I don't think it would be trivial for us to round the the coordinates, as we would have to do it for every bar, so it sort of compounds, eg:

```tsx
const bars = [
  {
    x: 1.45, // round down? 1 ok
    y: 10.6, // round up? 11 ok
  },
  {
    x: 11.20, // round down? 11, not ok
    y: 20.65, // round up? 21, ok
  },
];
```

Reply:
Why the ` x: 11.20, // round down? 11, not ok` is not ok?

If we round on `px` level it should not be an issue. It's either us who do the rounding or it's the SVG render engine, but in any case, they can not display a rectangle border at `x=6.34px`.

I tried a PR https://github.com/mui/mui-x/pull/12846 to see how it impacts argos. But if we do that we will need to go for more than just the bar, because otherwise, we could have a gap in (-0.5, 0.5) between the bar and the x-axis line

Reply:
Ah. nevermind, I though you meant the vertical lines lines between rectangles 😅 

![323868823-5223409f-5b67-4f57-9651-33f5da7a5a52](https://github.com/mui/mui-x/assets/9891233/59e96e30-1ea4-4b99-99b2-f6c8ebe36534)

Yeah, clamping those would likely work 🤔

Reply:
> we could have a gap in (-0.5, 0.5) between the bar and the x-axis line

You mean between the different types of graphs, like a point in a line graph would be in a wrong location? Yeah I see how it could complicate things.

Could we make it so only the rectangle below adjust its size to max/min depending on x/y sign(+/-) so the next bar starts on top of it?

If we manage to stack them correctly, no gap should be visible.

Reply:
> Could we make it so only the rectangle below adjusts its size to max/min depending on x/y sign(+/-) so the next bar starts on top of it?

It's a solution, but I realized my issue was not an important one, because rectangle are clip with a `clip-path` applied to the drawing area, and the axis are plot on top. So this 0.5 diff should not be visible